srv/salt/_modules/cephprocesses.py

```diff
@@ -12,27 +12,39 @@
 A secondary purpose is a utility to check the current state of all services.
 """
 
-def check():
+def check(roles=[]):
     """
     Query the status of running processes for each role.  Return False if any
     fail.
     """
-    processes = { 'mon': [ 'ceph-mon' ],
-                 'storage': [ 'ceph-osd' ],
-                 'mds': [ 'ceph-mds' ],
-                 'igw': [ 'lrbd' ],
-                 'rgw': [ 'radosgw' ],
-                 'ganesha': [ 'ganesha.nfsd', 'rpcbind', 'rpc.statd' ] }
+    processes = {'mon': ['ceph-mon'],
+                 'storage': ['ceph-osd'],
+                 'mds': ['ceph-mds'],
+                 'igw': ['lrbd'],
+                 'rgw': ['radosgw'],
+                 'ganesha': ['ganesha.nfsd', 'rpcbind', 'rpc.statd']}
 
     ret = True
+
+    def check_process(role):
+        for process in processes[role]:
+            result = __salt__['status.pid'](process)
+            log.info("Pid for process {} is {}".format(process, result))
+            if result == '':
```

Comment:
Salt hasn't proven itself to have stable internals. We should be prepared for Salt to return a string that has whitespace but no content, a `None` value, or error text.

I might suggest testing instead that the result is (or is not) in the form we expect of a pid, which is conveniently just a positive integer.

```py
if not pid.isdigit():
    log.error('...')
```